runners/core-java/src/main/java/org/apache/beam/runners/core/ReduceFnRunner.java

```diff
@@ -919,8 +920,9 @@ private boolean needToEmit(boolean isEmpty, boolean isFinished, PaneInfo.Timing
       // The pane has elements.
       return true;
     }
-    if (timing == Timing.ON_TIME) {
-      // This is the unique ON_TIME pane.
+    if (timing == Timing.ON_TIME
+        && windowingStrategy.getOnTimeBehavior() == Window.OnTimeBehavior.FIRE_ALWAYS) {
```

Comment:
For the panel which is `Timing.ON_TIME` and `isFinished`, `OnTimeBehavior.FIRE_ALWAYS` will hide the value of `ClosingBehavior`, I'm not sure if this is the expected working order for the two behavior settings, what do you think about that?